cmd/helm/completion.go

```diff
@@ -52,13 +54,14 @@ func newCompletionCmd(out io.Writer) *cobra.Command {
 
 	cmd := &cobra.Command{
 		Use:   "completion SHELL",
-		Short: "generate autocompletions script for the specified shell (bash or zsh)",
+		Short: "generate autocompletions script for the specified shell (bash, zsh or fish)",
 		Long:  completionDesc,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return runCompletion(out, cmd, args)
 		},
 		ValidArgs: shells,
 	}
+	cmd.Flags().BoolVar(&completionNoDesc, "no-descriptions", false, "disable completion descriptions for shells that support it")
```

Comment:
It was bothering me that this flag would apply to all shells instead of just the fish shell.
I've opened #8314 to allow to improve on this.  Let's see what the other maintainers think of this.

Reply:
Agreed! :+1: